wolfssl/wolfio.h

```diff
@@ -615,7 +615,82 @@ WOLFSSL_API void wolfSSL_SetIOWriteFlags(WOLFSSL* ssl, int flags);
     WOLFSSL_API   int wolfSSL_SetIO_LwIP(WOLFSSL* ssl, void *pcb,
                                 tcp_recv_fn recv, tcp_sent_fn sent, void *arg);
 #endif
+#ifdef WOLFSSL_ISOTP
+    #define ISOTP_DEFAULT_TIMEOUT 100
+    #define ISOTP_DEFAULT_WAIT_COUNT 3
+    #define ISOTP_DEFAULT_BUFFER_SIZE 16384
+    #define ISOTP_FIRST_FRAME_DATA_SIZE 6
+    #define ISOTP_SINGLE_FRAME_DATA_SIZE 7
+    #define ISOTP_MAX_CONSECUTIVE_FRAME_DATA_SIZE 7
+    #define ISOTP_MAX_MS_FRAME_DELAY 0x7f
+    #define ISOTP_CAN_BUS_PAYLOAD_SIZE 8
+    #define ISOTP_MAX_DATA_SIZE 4095
+    #define ISOTP_FLOW_CONTROL_PACKET_SIZE 3
+    #define ISOTP_FLOW_CONTROL_FRAMES 0 /* infinite */
+    #define ISOTP_MAX_SEQUENCE_COUNTER 15
+
+    enum isotp_frame_type {
+        ISOTP_FRAME_TYPE_SINGLE      = 0,
+        ISOTP_FRAME_TYPE_FIRST       = 1,
+        ISOTP_FRAME_TYPE_CONSECUTIVE = 2,
+        ISOTP_FRAME_TYPE_CONTROL     = 3
+    };
+
+    enum isotp_flow_control {
+        ISOTP_FLOW_CONTROL_CTS   = 0,
+        ISOTP_FLOW_CONTROL_WAIT  = 1,
+        ISOTP_FLOW_CONTROL_ABORT = 2
+    };
 
+    enum isotp_connection_state {
+        ISOTP_CONN_STATE_IDLE,
+        ISOTP_CONN_STATE_SENDING,
+        ISOTP_CONN_STATE_RECEIVING
+    };
+
+    typedef struct isotp_can_data {
+        byte data[8];
+        byte length;
+    } isotp_can_data;
+
+    /* User supplied functions for sending/receiving CAN bus messages of up to
+     * 8 bytes, as well as a function to add an artificial delay when a
+     * receiver requests one. */
+    typedef int (*can_recv_fn)(struct isotp_can_data *data, void *arg,
+            int timeout);
+    typedef int (*can_send_fn)(struct isotp_can_data *data, void *arg);
+    typedef void (*can_delay_fn)(int microseconds);
+
+    typedef struct isotp_wolfssl_ctx {
+        int socket;
```

Comment:
Have you considered sorting these by largest to smallest? Allows the most compact struct... Pointer, word32, word16, byte, enum, etc...

Reply:
Sure thing. Put the frame struct first since that is 9 bytes. Also `int socket` shouldn't be there any more, that was from the original test code, removing...